server/src/utils/adjustFollowingDeliveryPhases.js

```diff
@@ -0,0 +1,44 @@
+const { db } = require('../db/models');
+
+module.exports = (model) => {
+  const newModel = model;
+  const nextUnfrozenPhase = async () => {
+    if (newModel.itemInFrontOf) {
+      const nextPhase = await db.deliveryPhase.findOne({
+        where: { id: newModel.itemInFrontOf },
+      });
+      if (nextPhase.freezeDuration) {
+        if (nextPhase.itemInFrontOf) {
+          newModel.itemInFrontOf = nextPhase.itemInFrontOf;
+          nextUnfrozenPhase(newModel);
+        } else {
+          nextPhase.itemInFrontOf = newModel.id;
+          await db.deliveryPhase.upsert(nextPhase.dataValues);
+          newModel.itemInFrontOf = null;
+          await db.deliveryPhase.upsert(newModel);
+        }
+      }
+      if (!nextPhase.freezeDuration) {
```

Comment:
Wouldn't this be an else to your first if

Reply:
fixed